builder/builder.go

```diff
@@ -194,6 +198,10 @@ func hasBPWithVersion(bps []BuildpackMetadata, version string) bool {
 	return false
 }
 
+func (b *Builder) SetDescription(description string) {
```

Comment:
This is starting to look like Java. Why do we need getters/setters here? I see we're returning a concrete `*Builder` in the constructor. If we need to change these properties via an interface,  could we refactor to pull the state out into data structs?

Reply:
@sclevine This type follows a builder-esque (not to be confused with a CNB builder lol) pattern so is stateful. If you still think it should be changed, I'm not sure I'm following your proposed solution. Could you clarify?

Reply:
Per side conversation, I'll capture this in a chore and merge as is